Build issue with opam in llvm-15.0.0

The build fails with `The script must be executed from its current directory` in setup.sh. I added `cd $(dirname $0)` to the top of the file and removed the exit inside the check and it builds fine. Not sure if this is an issue with my setup.

Here is the output. I am inside a podman container with ubuntu 22.04 image, not sure if that makes a difference.
```
The following actions will be performed:
  - install conf-bash         1            [required by conf-llvm]
  - install dune              3.10.0       [required by llvm]
  - install conf-llvm         15           [required by llvm]
  - install stdlib-shims      0.3.0        [required by integers]
  - install csexp             1.5.2        [required by dune-configurator]
  - install bigarray-compat   1.1.0        [required by ctypes]
  - install integers          0.7.0        [required by ctypes]
  - install dune-configurator 3.10.0       [required by ctypes]
  - install ctypes            0.21.1       [required by llvm]
  - install llvm              15.0.7+nnp-2
===== 10 to install =====
Do you want to continue? [Y/n] y

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
-> retrieved bigarray-compat.1.1.0  (cached)
-> retrieved csexp.1.5.2  (cached)
-> retrieved ctypes.0.21.1  (cached)
+ /usr/bin/bash "--version" (CWD=/home/tanapoom/.opam/test-switch/.opam-switch/build/conf-bash.1)
- GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)
- Copyright (C) 2020 Free Software Foundation, Inc.
- License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
- 
- This is free software; you are free to change and redistribute it.
- There is NO WARRANTY, to the extent permitted by law.
- GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)
- Copyright (C) 2020 Free Software Foundation, Inc.
- License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
- 
- This is free software; you are free to change and redistribute it.
- There is NO WARRANTY, to the extent permitted by law.
-> compiled  conf-bash.1
-> installed conf-bash.1
-> retrieved integers.0.7.0  (cached)
+ /usr/bin/bash "configure.sh" "15" (CWD=/home/tanapoom/.opam/test-switch/.opam-switch/build/conf-llvm.15)
- + version=15
- + hash brew
- + shopt -s nullglob
- + for llvm_config in llvm-config-$version llvm-config-${version}.0 llvm-config${version}0 llvm-config-mp-$version llvm-config-mp-${version}.0 llvm${version}-config llvm-config-${version}-32 llvm-config-${version}-64 llvm-config $brew_llvm_config
- ++ llvm-config-15 --version
- + llvm_version=15.0.7
- + case $llvm_version in
- + echo 'config: "llvm-config-15"'
- + echo 'version: "15.0.7"'
- + exit 0
- + version=15
- + hash brew
- + shopt -s nullglob
- + for llvm_config in llvm-config-$version llvm-config-${version}.0 llvm-config${version}0 llvm-config-mp-$version llvm-config-mp-${version}.0 llvm${version}-config llvm-config-${version}-32 llvm-config-${version}-64 llvm-config $brew_llvm_config
- ++ llvm-config-15 --version
- + llvm_version=15.0.7
- + case $llvm_version in
- + echo 'config: "llvm-config-15"'
- + echo 'version: "15.0.7"'
- + exit 0
-> compiled  conf-llvm.15
-> installed conf-llvm.15
-> retrieved dune.3.10.0  (cached)
-> retrieved dune-configurator.3.10.0  (cached)
-> retrieved stdlib-shims.0.3.0  (cached)
-> retrieved llvm.15.0.7+nnp-2  (https://opam.ocaml.org/cache)
+ /home/tanapoom/.opam/test-switch/bin/ocaml "boot/bootstrap.ml" "-j" "15" (CWD=/home/tanapoom/.opam/test-switch/.opam-switch/build/dune.3.10.0)
- ocamlc -output-complete-exe -w -24 -g -o .duneboot.exe -I boot -I +unix unix.cma boot/libs.ml boot/duneboot.ml
- ./.duneboot.exe -j 15
- 
+ /home/tanapoom/.opam/test-switch/.opam-switch/build/dune.3.10.0/./_boot/dune.exe "build" "dune.install" "--release" "--profile" "dune-bootstrap" "-j" "15" (CWD=/home/tanapoom/.opam/test-switch/.opam-switch/build/dune.3.10.0)
-> compiled  dune.3.10.0
-> installed dune.3.10.0
+ /home/tanapoom/.opam/test-switch/bin/dune "build" "-p" "stdlib-shims" "-j" "15" (CWD=/home/tanapoom/.opam/test-switch/.opam-switch/build/stdlib-shims.0.3.0)
-> compiled  stdlib-shims.0.3.0
-> installed stdlib-shims.0.3.0
+ /home/tanapoom/.opam/test-switch/bin/dune "build" "-p" "csexp" "-j" "15" "@install" (CWD=/home/tanapoom/.opam/test-switch/.opam-switch/build/csexp.1.5.2)
-> compiled  csexp.1.5.2
-> installed csexp.1.5.2
+ /home/tanapoom/.opam/test-switch/bin/dune "build" "-p" "bigarray-compat" "-j" "15" (CWD=/home/tanapoom/.opam/test-switch/.opam-switch/build/bigarray-compat.1.1.0)
-> compiled  bigarray-compat.1.1.0
-> installed bigarray-compat.1.1.0
+ /home/tanapoom/.opam/test-switch/bin/dune "build" "-p" "integers" "-j" "15" (CWD=/home/tanapoom/.opam/test-switch/.opam-switch/build/integers.0.7.0)
-> compiled  integers.0.7.0
-> installed integers.0.7.0
+ /usr/bin/rm "-rf" "vendor/csexp" (CWD=/home/tanapoom/.opam/test-switch/.opam-switch/build/dune-configurator.3.10.0)
+ /usr/bin/rm "-rf" "vendor/pp" (CWD=/home/tanapoom/.opam/test-switch/.opam-switch/build/dune-configurator.3.10.0)
+ /home/tanapoom/.opam/test-switch/bin/dune "build" "-p" "dune-configurator" "-j" "15" "@install" (CWD=/home/tanapoom/.opam/test-switch/.opam-switch/build/dune-configurator.3.10.0)
-> compiled  dune-configurator.3.10.0
-> installed dune-configurator.3.10.0
+ /home/tanapoom/.opam/test-switch/bin/dune "build" "-p" "ctypes" "-j" "15" "--promote-install-files=false" "@install" (CWD=/home/tanapoom/.opam/test-switch/.opam-switch/build/ctypes.0.21.1)
+ /home/tanapoom/.opam/test-switch/bin/dune "install" "-p" "ctypes" "--create-install-files" "ctypes" (CWD=/home/tanapoom/.opam/test-switch/.opam-switch/build/ctypes.0.21.1)
-> compiled  ctypes.0.21.1
-> installed ctypes.0.21.1
+ /home/tanapoom/.opam/test-switch/.opam-switch/build/llvm.15.0.7+nnp-2/./setup.sh "llvm-config-15" (CWD=/home/tanapoom/.opam/test-switch/.opam-switch/build/llvm.15.0.7+nnp-2)
- + dirname /home/tanapoom/.opam/test-switch/.opam-switch/build/llvm.15.0.7+nnp-2/./setup.sh
- + test /home/tanapoom/.opam/test-switch/.opam-switch/build/llvm.15.0.7+nnp-2/. != .
- + echo The script must be executed from its current directory.
- The script must be executed from its current directory.
- + exit 1
[ERROR] The compilation of llvm.15.0.7+nnp-2 failed at "./setup.sh llvm-config-15".

#=== ERROR while compiling llvm.15.0.7+nnp-2 ==================================#
# context     2.1.2 | linux/x86_64 | ocaml-base-compiler.5.0.0 | https://opam.ocaml.org#ba404223
# path        ~/.opam/test-switch/.opam-switch/build/llvm.15.0.7+nnp-2
# command     ~/.opam/test-switch/.opam-switch/build/llvm.15.0.7+nnp-2/./setup.sh llvm-config-15
# exit-code   1
# env-file    ~/.opam/log/llvm-510774-e2da2d.env
# output-file ~/.opam/log/llvm-510774-e2da2d.out
### output ###
# + dirname /home/tanapoom/.opam/test-switch/.opam-switch/build/llvm.15.0.7+nnp-2/./setup.sh
# + test /home/tanapoom/.opam/test-switch/.opam-switch/build/llvm.15.0.7+nnp-2/. != .
# + echo The script must be executed from its current directory.
# The script must be executed from its current directory.
# + exit 1



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+- The following actions failed
| - build llvm 15.0.7+nnp-2
+- 
+- The following changes have been performed
| - install bigarray-compat   1.1.0
| - install conf-bash         1
| - install conf-llvm         15
| - install csexp             1.5.2
| - install ctypes            0.21.1
| - install dune              3.10.0
| - install dune-configurator 3.10.0
| - install integers          0.7.0
| - install stdlib-shims      0.3.0
+- 

The former state can be restored with:
    /usr/bin/opam switch import
"/home/tanapoom/.opam/test-switch/.opam-switch/backup/state-20230915082615.export"
'opam install llvm --verbose' failed.
```

--- a/setup.sh
+++ b/setup.sh
@@ -3,7 +3,7 @@
 set -e
 set -x
 
-if test "$(dirname $0)" != '.'; then
+if test ! "$(dirname $0)" -ef '.'; then
     echo "The script must be executed from its current directory."
     exit 1
 fi

